batch/batch/worker/worker.py

```diff
@@ -1126,6 +1126,25 @@ async def container_config(self):
             'CAP_KILL',
             'CAP_AUDIT_WRITE',
         ]
+
+        nvidia_runtime_hook = []
+        if 'g2' in INSTANCE_CONFIG["machine_type"]:
+            log.info("g2-machine")
+            nvidia_runtime_hook = [
+                {
+                    "path": "/usr/bin/nvidia-container-runtime-hook",
+                    "args": ["nvidia-container-runtime-hook", "prestart"],
+                    "env": [
+                        "LANG=C.UTF-8",
+                        "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin",
+                        "NOTIFY_SOCKET=/run/systemd/notify",
+                        "INVOCATION_ID=8b6eb73e784f439397beb9fd9b9aaba5",
+                        "JOURNAL_STREAM=8:17167",
+                        "SYSTEMD_EXEC_PID=761",
```

Comment:
I don't think the environment variables with hardcoded numbers make sense here.  I'd try removing INVOCATION_ID, JOURNAL_STREAM and SYSTEMD_EXEC_PID.